### PR TITLE
Global Styles: Add support for caption elements

### PR DESCRIPTION
**Pull request by scruffian**
## What?
This makes it possible to style captions in theme.json by adding this into your theme.json file:

```
{
	"styles": {
		"elements": {
			"caption": {
				"color": {
					"background": "red",
					"text": "yellow"
				}
			}
		}
	}
}
```

## Why?
So that themes can independently control the display of their captions. See https://github.com/WordPress/gutenberg/issues/40205

## How?
We create a new element called caption and then add the `wp-element-caption` class to all elements that we want to use the caption styles.

The problem with this approach is that captions are also targeted by the default styles in the block themselves. This is not a problem as the CSS global styles outputs is stronger, except for the case of captions inside gallery blocks.

## Testing Instructions
- Add the snippet above to your theme
- Add an audio/image/table/embed/video/gallery block to the page
- Confirm that the captions have the styles added above.

## Screenshots or screencast 
Screenshot 2022-06-16 at 14 00 22

### CODE REVIEW

**Comment by carolinan** [comment]
If we choose this path, does it mean that we are no longer considering making the captions separate inner blocks?

**Comment by scruffian** [comment]
> If we choose this path, does it mean that we are no longer considering making the captions separate inner blocks?

I believe so yes. Inner blocks bring a lot of extra complexity with them (for example the block toolbar and inspector controls) which may not give us the best user experience for editing the block they are part of.

**Comment by scruffian** [comment]
cc @WordPress/block-themers

**Review by draganescu — APPROVED** [review]
I believe this follows in the footsteps of adding the button element. 🚢 it.

@carolinan it's not known. We're adding element support so that `theme.json` can style this element, but if we arrive at a place where a `Caption` block makes sense it will not be blocked by the element, but it will probably use it and inherit the styling seamlessly.

**Comment by ndiego** [comment]
@scruffian, this seems to work in the Editor, but not 100% on the Frontend. In my testing, certain values are overridden by certain block styles: 
image

**Comment by bgardner** [comment]
@scruffian +1 to @ndiego's comment. Seems similar to this: https://github.com/WordPress/gutenberg/pull/41934

**Comment by scruffian** [comment]
We have an issue for this: https://github.com/WordPress/gutenberg/issues/42054

**Comment by mburridge** [comment]
Added the Needs Dev Note label in case this needs a dev note (either individual or as part of a "misc" dev note) for WP 6.1 release.

**Comment by scruffian** [comment]
I think this is covered by: https://make.wordpress.org/core/?p=98557

**Comment by mburridge** [comment]
Thanks @scruffian, I've removed the label.

File: packages/block-library/src/audio/save.js [hunk 1]
```diff
@@ -17,7 +21,13 @@ export default function save( { attributes } ) {
 					preload={ preload }
 				/>
 				{ ! RichText.isEmpty( caption ) && (
-					<RichText.Content tagName="figcaption" value={ caption } />
+					<RichText.Content
+						tagName="figcaption"
+						value={ caption }
+						className={ __experimentalGetElementClassName(
```
Comment (youknowriad) [inline packages/block-library/src/audio/save.js:27]: Any change to a save function of a block need a deprecation to be aded to that block. See https://github.com/WordPress/gutenberg/issues/45153

**Pull request merged**